components/filter/filter.js

```diff
@@ -0,0 +1,259 @@
+			</d2l-hierarchical-view>
+		`;
+	}
+
+	_buildDimensions() {
+		return Array.from(this._dimensions.values(), (dimension) => {
+			const builtDimension = this._buildDimension(dimension);
+			return html`<d2l-menu-item text="${dimension.text}">
+				${builtDimension}
+			</d2l-menu-item>`;
+		});
+	}
+
+	_createListDimension(dimension) {
+		const slot = dimension.shadowRoot.querySelector('slot');
+		const items = slot.assignedNodes().filter(node => node.nodeType ===  Node.ELEMENT_NODE);
+		return html`
+			<d2l-list
+				@d2l-list-selection-change="${this._handleChangeListDimension}"
+				extend-separators>
+
+				${items.map(item => html`
+					<d2l-list-item
+						key="${item.key}"
+						selectable
+						?selected="${item.selected}"
+						slim>
+
+						<div class="d2l-filter-dimension-value-text">${item.text}</div>
+					</d2l-list-item>
+				`)}
+			</d2l-list>
+		`;
+	}
+
+	_dispatchChangeEvent(eventDetail) {
+		this.dispatchEvent(new CustomEvent('d2l-filter-change', { bubbles: false, composed: false, detail: eventDetail }));
+	}
+
+	_handleChangeListDimension(e) {
+		const dimensionKey = e.composedPath()[0].parentNode.id;
+		const valueKey = e.detail.key;
+		const selected = e.detail.selected;
+
+		// Update the corresponding d2l-filter-dimension-value to keep them in sync
+		const dimension = this._dimensions.get(dimensionKey);
+		const slot = dimension.shadowRoot.querySelector('slot');
+		const items = slot.assignedNodes().filter(node => node.nodeType ===  Node.ELEMENT_NODE);
+		const item = items.find(item => item.key === valueKey);
+		item.selected = selected;
+
+		this._dispatchChangeEvent({ dimension: dimensionKey, value: { key: valueKey, selected: selected } });
+	}
+
+	_handleDataChangeListDimension(e) {
+		const dimension = this.shadowRoot.getElementById(e.target.key);
+		const changes = e.detail.changes;
+		if (changes.has('text')) {
+			const menuItem = dimension.parentNode;
+			if (menuItem.text !== changes.get('text')) {
+				this.requestUpdate();
+				return;
+			}
+		}
+	}
+
+	_handleDataChangeListDimensionValue(e) {
+		const dimension = this.shadowRoot.getElementById(e.target.parentNode.key);
+		const value = dimension.querySelector(`[key="${e.target.key}"]`);
+		const changes = e.detail.changes;
+
+		if (changes.has('text')) {
+			const text = value.querySelector('.d2l-filter-dimension-value-text');
+			if (text.innerText !== changes.get('text')) {
+				text.innerText = changes.get('text');
+			}
+		}
+		if (changes.has('selected')) {
+			if (value.selected !== changes.get('selected')) {
+				value.selected = changes.get('selected');
+			}
+		}
+	}
+
+	_handleDimensionHide() {
+		this.shadowRoot.querySelector(`#${this._activeDimensionKey}`).hide();
+		this._activeDimensionKey = null;
+	}
+
+	_handleDimensionShowComplete() {
+		const dimension = this.shadowRoot.querySelector(`#${this._activeDimensionKey}`);
+		const focusable = getFirstFocusableDescendant(dimension);
+		if (focusable) {
+			focusable.focus();
+		}
```

Comment:
This function is currently setup to be used for every dimension type so it is very generic in finding the focus element.  But we could create a custom one for each type if this feels "bad"